### PR TITLE
LPS-145189 SVG can not be previewed

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
https://issues.liferay.com/browse/LPS-145189

Regression caused by LPS-144830.

The root cause of this issue is:
 - LPS-144830 starts checking the size of converted files
 - But SVG files are not converted, we return a fake SVGAdaptiveMedia that returns the original fileVersion.getContentStream correctly, but it doesn't return the file atributes
 - As it doesn't have any attribute, size zero is returned.

To fix this problem, I am coping the *_createRawAdaptiveMedia* method from AMImageRequestHandler that is correctly handling the FileVersion object

I have move duplicated code to the AMImageAttributeMapping class, let me know if there is a better place to put it.

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
After reviewing the ci tests results, it seems there are two NPEs with the new code, I am going to review them.

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/configuration/AMImageAttributeMapping.java [hunk 1]
```diff
@@ -75,6 +76,22 @@ public static AMImageAttributeMapping fromProperties(
...
+				Optional.of(fileVersion.getSize())
+			).put(
+				AMAttribute.getContentTypeAMAttribute(),
+				Optional.of(fileVersion.getMimeType())
+			).put(
+				AMAttribute.getFileNameAMAttribute(),
+				Optional.of(fileVersion.getFileName())
+			).build());
+	}
+
```
Comment (adolfopa) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/configuration/AMImageAttributeMapping.java:94]: Can you rename this method to `fromFileVersion()`?

**Review by adolfopa — CHANGES_REQUESTED** [review]
Other than the minor rename, this looks good :+1: Thanks!

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/configuration/AMImageAttributeMapping.java (hunk 1 above)
Comment (jorgediaz-lr) [reply modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/configuration/AMImageAttributeMapping.java:94]: Yes, no problem

**Comment by jorgediaz-lr** [comment]
@adolfopa I have done following changes:

- https://github.com/liferay-lima/liferay-portal/pull/2643/commits/250f8256bba06adf72bf389dcf22a9b3c1139772 => I have updated this commit to replace the field name `fromRawFileVersion` to `fromFileVersion`
- https://github.com/liferay-lima/liferay-portal/pull/2643/commits/6562cfbe8ebd7d59977dffd9b60920288385063c and https://github.com/liferay-lima/liferay-portal/pull/2643/commits/b8efc6021e288e3e80e571e239d5ada68b9c93b7 => new commits to solve the CI test errors
- https://github.com/liferay-lima/liferay-portal/pull/2643/commits/fe3dec093467021a4b3773d06ced309dbb123cca => I have extended the **AMImageFinderImplTest.testGetSVGMediaInputStream** test to also check the file size (this was the root cause of the issue)

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**